system/Cache/Handlers/BaseHandler.php

```diff
@@ -14,12 +14,65 @@
 use Closure;
 use CodeIgniter\Cache\CacheInterface;
 use Exception;
+use InvalidArgumentException;
 
 /**
  * Base class for cache handling
  */
 abstract class BaseHandler implements CacheInterface
 {
+	/**
+	 * Reserved characters that cannot be used in a key or tag.
+	 *
+	 * @see https://github.com/symfony/cache-contracts/blob/c0446463729b89dd4fa62e9aeecc80287323615d/ItemInterface.php#L43
+	 */
+	public const RESERVED_CHARACTERS = '{}()/\@:';
+
+	/**
+	 * Maximum key length.
+	 */
+	public const MAX_KEY_LENGTH = PHP_INT_MAX;
+
+	/**
+	 * Prefix to apply to cache keys.
+	 * May not be used by all handlers.
+	 *
+	 * @var string
+	 */
+	protected $prefix;
+
+	/**
+	 * Validates a cache key according to PSR-6.
+	 * Keys that exceed MAX_KEY_LENGTH are hashed.
+	 *
+	 * @param string $key    The key to validate
+	 * @param string $prefix Optional prefix to include in length calculations
+	 *
+	 * @throws InvalidArgumentException When $key is not valid
+	 *
+	 * @see https://github.com/symfony/cache/blob/7b024c6726af21fd4984ac8d1eae2b9f3d90de88/CacheItem.php#L158
+	 */
+	public static function validateKey($key, $prefix = ''): string
+	{
+		if (! is_string($key))
+		{
+			throw new InvalidArgumentException('Cache key must be a string');
+		}
+		if ($key === '')
+		{
+			throw new InvalidArgumentException('Cache key cannot be empty.');
```

Comment:
I'm not necessarily opposed to throwing the exception directly here, but most of the rest of the framework uses external exception classes with static methods. Does it make sense to do that here for consistency or is that overkill?

Reply:
Your two comments are intimately related. Throughout the framework we don't localize `InvalidArgumentException` content precisely because they are not runtime exceptions (and thus should use the `new static FrameworkException` approach).

That said, I think these are up for grabs as to whether they should be considered invalid arguments or runtime issues. I went with `InvalidArgumentException` because these will be re-used with the PSR-6 adapter which requires a cache-specific `InvalidArgumentException` for precisely these failures: https://www.php-fig.org/psr/psr-6/

Reply:
That's fair. I'm fine with this then.